### DISCUSSION TITLE
i want to know step by step procedure to use API to get data in CAP format

### DISCUSSION THREAD

**Discussion by Avanishchaurasiya** [weather-gov/api#364] [Q&A]
**NOTE**
Please do not report operational issues (such as outages) here. We can not support these issues via this forum. Please report these as described here: https://weather-gov.github.io/api/reporting-issues

**Please describe what you're trying to do.**
A clear and concise description of what your issue is. Ex. I'm looking for forecast data...

**Comment by scadergit** [comment]
This made me look back over the [Examples tab on the documentation page](https://www.weather.gov/documentation/services-web-api) and realize the Alerts example is missing details on the common CAP format. I'll add those shortly. Until then...

The most important step in requesting CAP is setting [the request Accept header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept). Alert endpoints either return a list of alert summaries, or details of a single alert. The lists use ATOM, and the details use CAP. You can also use the default JSON to discover then request CAP for specific alerts. 

To return a list in ATOM, the request should use the Accept header `application/atom+xml`. Lists include all /alerts endpoints except the /alerts/{id} endpoint. To return details of a specific alert in CAP, the request should use the Accept header `application/cap+xml`. Details are only returned for a specific alert using the /alerts/{id} endpoint.

If you already know the alert id, you can directly request the /alerts/{id} endpoint. For example (note this id will expire at some point and be invalid):
`https://api.weather.gov/alerts/NWS-IDP-PROD-4136017-3504489`

Here is an example of using an Accept header for CAP in curl:
`curl -H "Accept: application/cap+xml" https://api.weather.gov/alerts/NWS-IDP-PROD-4136017-3504489`

The other /alerts endpoints and parameters can be used to discover alert ids, including active alerts using the /alerts/active endpoint. For example:
`https://api.weather.gov/alerts/active`

Here is an example of using an Accept header for ATOM in curl:
`curl -H "Accept: application/atom+xml" https://api.weather.gov/alerts/active`
